sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -78,29 +78,66 @@ class SessionCatalog(
     if (conf.caseSensitiveAnalysis) name else name.toLowerCase
   }
 
+  /**
+   * Validate names
+   */
+  protected[this] def validateName(name: String): Boolean = {
+    val validName = "([_A-Za-z0-9]*)".r
```

Comment:
Hmm. Mine also seems not enough since it still allows a strange name like `___`. I'm sure you'll find a good pattern for this. :)